plugin/src/main/scala/chisel3/internal/plugin/BundleComponent.scala

```diff
@@ -63,62 +97,177 @@ private[plugin] class BundleComponent(val global: Global, arguments: ChiselPlugi
+                } else {
+                  //TODO: The following code block creates a compiler error for bundle fields that are
+                  //      Seq[Data]. The test for this case is a bit too hacky so for now we will just
+                  //      elide these fields.
+//                  if (member.isAccessor && typeIsSeqOfData(member.tpe) && !isIgnoreSeqInBundle(bundleSymbol)) {
+//                    global.reporter.error(
+//                      member.pos,
+//                      s"Bundle.field ${bundleSymbol.name}.${member.name} cannot be a Seq[Data]. " +
+//                        "Use Vec or MixedVec or mix in trait IgnoreSeqInBundle"
+//                    )
+//                  }
+                  None
+                }
+
+              case _ => None
+            }.toList
+
+            indentShow(s"\nstarting parents: ${bundleSymbol.parentSymbols.mkString(",")}")
+            val allParentFields = bundleSymbol.parentSymbols.flatMap { parentSymbol =>
+              val fieldsFromParent = if (depth < 1 && !isExactBundle(bundleSymbol)) {
+                indentShow(s"Calling getAll($parentSymbol) ")
+                val foundFields = getAllBundleFields(parentSymbol, depth + 1)
+                indentShow(s"Return getAll($parentSymbol): " + foundFields.map(_._1).mkString(","))
+                foundFields
+              } else {
+                List()
+              }
+              fieldsFromParent
+            }
+            allParentFields ++ currentFields
+          }
+
+          val elementArgs = getAllBundleFields(bundle.symbol)
+
+          val elementsImplSym =
+            bundle.symbol.newMethod(TermName("_elementsImpl"), bundle.symbol.pos.focus, Flag.OVERRIDE | Flag.PROTECTED)
+          // Handwritten cloneTypes don't have the Method flag set, unclear if it matters
+          elementsImplSym.resetFlag(Flags.METHOD)
+          // Need to set the type to chisel3.Bundle for the override to work
+          elementsImplSym.setInfo(NullaryMethodType(seqMapTpe))
+
+          val elementsImpl = localTyper.typed(
+            DefDef(elementsImplSym, q"scala.collection.immutable.SeqMap.apply[String, Any](..$elementArgs)")
+          )
+
+          //TODO: These should go away
+          show("ELEMENTS:\n" + elementArgs.map { case (symbol, tree) => s"($symbol, $tree)" }.mkString("\n"))
+          additionalMethods ++= Seq(elementsImpl)
+
+          show(("#" * 80 + "\n") * 2)
+        }
 
         // ==================== Generate _usingPlugin ====================
         // Unclear why quasiquotes work here but didn't for cloneTypeSym, maybe they could.
         val usingPlugin = localTyper.typed(q"override protected def _usingPlugin: Boolean = true")
+        additionalMethods ++= Seq(usingPlugin)
 
         val withMethods = deriveClassDef(bundle) { t =>
-          deriveTemplate(t)(_ :+ cloneTypeImpl :+ usingPlugin)
+          deriveTemplate(t)(_ ++ additionalMethods)
```

Comment:
Note that instead of `var additionalMethods`, if you make `elementsImpl` be of type `Option[Tree]`, this line can just be:
```scala
deriveTemplate(t)(_ :+ cloneTypeImpl :+ usingPlugin ++ elementsImpl)
```

Reply:
Yes, this is will be cleaner
Done